pkg/helm/helm.go

```diff
@@ -84,6 +86,7 @@ type LoadChartConfig struct {
 	CertFile string `yaml:"certFile,omitempty"`
 	KeyFile  string `yaml:"keyFile,omitempty"`
 	CaFile   string `yaml:"caFile,omitempty"`
+	LockFile string `yaml:"lockFile,omitempty"`
```

Comment:
Why specify the LockFile explicitly when a Chart.yaml or requirements.yaml does not?

Maybe the `requirements.yaml` could be written into a file named like the generator yaml suffixed with `.lock` if it is not a local chart. WDYT?
Thinking about the issues here

Reply:
> Let's simply use helm's standard lock file location to be consistent and compatible with helm and keep the interface simple.

Where is that?

> People who want to use this feature could specify a local chart (though it requires more writing).

This would e.g. require an additional level of indication in `values` files; in addition to an extra subfolder just containing Chart.yaml, requirements.yaml and requirements.lock. I recall some issue with tooling when you have an empty Chart.yaml without a version field in it too... so you may have to invent some abitrary version for your local indirection chart.
I don't think this is a great solution.

Reply:
With standard location I meant the local helm chart's `requirements.lock`. I must admit specifying the local helm chart is a bit cumbersome. You could set some valid static value as the Chart.yaml's version.

Reply:
@james-callahan as @scjudd pointed out `requirements.lock` specifies the hash of `requirements.yaml` - nothing more:
* https://github.com/mgoltzsche/helm-kustomize-plugin/blob/c371f81d721b1de202d9b0b88e2355bd7dfa06aa/vendor/k8s.io/helm/pkg/resolver/resolver.go#L150
* https://github.com/mgoltzsche/helm-kustomize-plugin/blob/c371f81d721b1de202d9b0b88e2355bd7dfa06aa/vendor/k8s.io/helm/pkg/provenance/sign.go#L405

@scjudd why specify the lockfile explicitly although we need it only when a `requirements.yaml` exists for which we could use the standard path `requirements.lock`?